### DISCUSSION TITLE
Is it possible to switch between versions via the cli?

### DISCUSSION THREAD

**Discussion by RobiNN1** [leokhoa/laragon#412] [Q&A]
As title says. Not sure if lagaron support cli but if so how?  

I have one project where unfortunately I have to use old php 5.6 with old apache and nodejs. And wehen I switch between projects, I always need to switch these versions too. Can this be done with one command?

**Comment by Pen-y-Fan** [comment]
I'm not sure about switching using CLI. I've downloaded the portable version for specific (old) projects.

![Laragon portible](https://user-images.githubusercontent.com/40126936/219775047-64333f9a-2029-410a-b56f-8491e9d15ecb.png)

If the full version is in C:\laragon, then add the portable in C:\laragon-portable. 

- Copy the required PHP directory from C:\laragon\bin\php\... to C:\laragon-portable\bin\php\...
- Copy the required Apache directory from C:\laragon\bin\apache\... to C:\laragon-portable\bin\apache\...
- Copy the required Node directory from C:\laragon\bin\node\... to C:\laragon-portable\bin\node\...

Repeat for any other components (such as MySQL).

Start the portable version as required by selecting c:\laragon-portable\laragon.exe

![Laragon portible](https://user-images.githubusercontent.com/40126936/219776015-d7452821-70b6-45f5-b06c-9638298ecec8.png)

The portable and full versions can not be run simultaneously unless the ports are changed, to avoid conflicts. E.g. the portable version uses ports 8080 and 33060. 

Optionally the required stack can be configured between the full and portable versions. For example, in the portable version, configure MySQL **not** to start and only start MySQL in the full version, to use that database server. That does undo some of the 'ease of starting' an old project but can save porting the database. Swings and roundabouts 🤷

![Laragon full and portable](https://user-images.githubusercontent.com/40126936/219783910-90cea1f7-a6d2-4417-b99c-0ec780e59f55.png)

\* This particular combination probably will not work as PHP 5.4.9 isn't compatible with MySQL 8.0, but does show the stack started, for illustration purposes.

The **laragon.exe** can be copied from C:\laragon\laragon.exe to C:\laragon-portable\laragon.exe too.

**Comment by RobiNN1** [comment]
I get it but I don't need a separate instance of laragon, everything works fine now. I've been using laragon for a while so I know these tricks to run multiple php versions at the same time. But this does not solve switching nodejs version.
It might be faster to use docker with particular setup so I don't have to keep changing versions in laragon.

Anyway, thanks for the reply, maybe it will help someone.

--
Support for such a feature will be really helpful. Sometimes it's faster to use a cli than a ui.

**Comment by Ayesh** [comment]
As far as I know, laragon has no CLI. I also think it'd be pretty useful if there was one.

**Reply by Pen-y-Fan** [reply]
Yes, there is a [Command Line](https://laragon.org/docs/cli.html), but the documentation only shows reloading Nginx and Apache.

Current CLI:

```shell
%laragon_root%\laragon reload nginx
%laragon_root%\laragon reload apache
```

It would need to `stop all`, `switch php 5.6` with `switch apache ...` and `switch node ...`, then `start all`

**Comment by rescenic** [comment]
I use NVM for Windows to switch Node.js version. You need to run cmd/pwsh as Admin.
[![laragon-nvm.png](https://i.postimg.cc/6q72f369/laragon-nvm.png)](https://postimg.cc/QK3dXjWP)

Download: https://github.com/coreybutler/nvm-windows

**Comment by Brugman** [comment]
This would be really useful. One of the last annoyances I have with Laragon is switching PHP versions. It rarely works without Stop / Start, but if I could automate the switch with a batch or AutoHotkey script that would circumvent the problem.
